packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1241,9 +1251,14 @@ export class InjectedScript {
           received: hasAttribute ? 'attribute present' : 'attribute not present',
         };
       } else if (expression === 'to.be.checked') {
-        result = this.elementState(element, 'checked');
-      } else if (expression === 'to.be.unchecked') {
-        result = this.elementState(element, 'unchecked');
+        const { checked, indeterminate } = options.expectedValue;
+        if (indeterminate) {
+          if (checked !== undefined)
+            throw this.createStacklessError('Checked state is not defined for indeterminate checkboxes');
```

Comment:
I'd rephrase it to make it clear that `checked` option cannot be specified along with `indeterminate`.